compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp

```diff
@@ -161,30 +120,24 @@ void TaggedShapeAnalyzer::analyze_transpose(const luci::CircleTranspose *transpo
  *              - value :   (1)   (448)  (49)
  *              - tags  :   (-)   (2)    (0, 1)
  */
-template <loco::DataType ReshapeType>
 bool TaggedShapeAnalyzer::analyze_reshape(const luci::CircleReshape *reshape_node)
 {
-  const luci::CircleConst *shape_node = loco::must_cast<luci::CircleConst *>(reshape_node->shape());
-  assert(shape_node->dtype() == ReshapeType);
-
   // At least one element must be in reshape's output-tensor.
-  if (shape_node->size<ReshapeType>() <= 0)
+  if (reshape_node->rank() <= 0)
```

Comment:
maybe above comment be updated ?

Reply:
without updating, the above comment still makes sense. 
So, I intentionally didn't update the comment.

Reply:
If I missed sth, please let me know :)

Reply:
OK ... but it's a bit hard to understand the meaning of the "element"
maybe revise this next time.

Reply:
> but it's a bit hard to understand the meaning of the "element" 

Oh, I see. Thank you for sharing. 
I'll fix it in another PR.